bson/src/test/unit/org/bson/types/ObjectIdTest.java

```diff
@@ -146,33 +147,32 @@ public void testToHexString() {
                 new ObjectId(Integer.MAX_VALUE, Short.MAX_VALUE, Short.MAX_VALUE, Short.MAX_VALUE).toHexString());
     }
 
-    private void checkDates(final String dateInTime, final ObjectId oid) {
-        SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy HH:mm:ssZ");
-        try {
-            Date d = dateFormat.parse(dateInTime);
-            assertEquals(oid.getDate(), d);
-        } catch (Exception e) {
-        }
-    }
-
     @Test
-    public void testTimeZero() {
-        checkDates("01-Jan-1970 00:00:00", new ObjectId(0, 0));
+    public void testTimeZero() throws ParseException {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MMM-yyyy HH:mm:ss Z");
```

Comment:
Introduce to getDate method that takes a string and returns a Date (If you using Extract Method refactoring, it will suggest that to you and find all the duplicates and replace them with the new method.  Pretty cool).

Reply:
Done. And, yes, very cool...